### PR TITLE
fix(firebase): validate custom `serverFunctionName`

### LINKED ISSUE #2643: Nuxt3 fails to build with preset when serverFunctionName is set
### Environment

Node version v20.15.0
Nuxt 3.12.4
Nitropack 2.9.7

### Reproduction

Create a new Nuxt3 project and try to build it with `npm run build -- --preset=firebase`

nuxt.config.ts
```
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  compatibilityDate: '2024-04-03',
  devtools: { enabled: true },
  nitro: {
    preset: "firebase",
    firebase: {
      gen: 2,
      serverFunctionName: "my-awesome-function",
      nodeVersion: "18",
      httpsOptions: {
        region: 'europe-west1',
      }
    }
  },
})
```

### Describe the bug

When building with preset firebase in latest version of Nuxt3 nitropack fails if serverFunctionName is set.

### Additional context

_No response_

### Logs

```sh
RollupError: node_modules/.pnpm/nitropack@2.9.7_magicast@0.3.4/node_modules/nitropack/dist/runtime/entries/firebase-gen-2.mjs (7:13): 'const' declarations must be initialized (Note that you need plugins to import files that are not JavaScript)


5: import { useAppConfig } from "#internal/nitro";
6: const firebaseConfig = useAppConfig().nitro.firebase;
7: export const __firebaseServerFunctionName__ = onRequest(
                ^
8:   {
9:     // Must be set to public to allow all public requests by default


[18:15:10]  ERROR  node_modules/.pnpm/nitropack@2.9.7_magicast@0.3.4/node_modules/nitropack/dist/runtime/entries/firebase-gen-2.mjs (7:13): 'const' declarations must be initialized (Note that you need plugins to import files that are not JavaScript)

  at getRollupError (/F:/tmp/test/node_modules/.pnpm/rollup@4.20.0/node_modules/rollup/dist/es/shared/parseAst.js:392:41)
  at ParseError.initialise (/F:/tmp/test/node_modules/.pnpm/rollup@4.20.0/node_modules/rollup/dist/es/shared/node-entry.js:11485:28)
  at convertNode (/F:/tmp/test/node_modules/.pnpm/rollup@4.20.0/node_modules/rollup/dist/es/shared/node-entry.js:13181:10)
  at convertProgram (/F:/tmp/test/node_modules/.pnpm/rollup@4.20.0/node_modules/rollup/dist/es/shared/node-entry.js:12525:12)
  at Module.setSource (/F:/tmp/test/node_modules/.pnpm/rollup@4.20.0/node_modules/rollup/dist/es/shared/node-entry.js:14346:24)
  at async ModuleLoader.addModuleSource (/F:/tmp/test/node_modules/.pnpm/rollup@4.20.0/node_modules/rollup/dist/es/shared/node-entry.js:19003:13)



[18:15:10]  ERROR  node_modules/.pnpm/nitropack@2.9.7_magicast@0.3.4/node_modules/nitropack/dist/runtime/entries/firebase-gen-2.mjs (7:13): 'const' declarations must be initialized (Note that you need plugins to import files that are not JavaScript)
```

### PR DESCRIPTION
**Pull request by jpsc**
### 🔗 Linked issue

Resolves #2643

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Normalize `serverFunctionName` to using underscore (`_`) instead of dashes (`-`). 

Resolves #2643

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/presets/firebase/preset.ts [hunk 1]
```diff
@@ -33,6 +33,12 @@ const firebase = defineNitroPreset(
         nitro.options.appConfig.nitro = nitro.options.appConfig.nitro || {};
         nitro.options.appConfig.nitro.firebase = nitro.options.firebase;
 
+        if(nitro.options.firebase?.serverFunctionName.includes('-')){
+          nitro.logger.warn(
```
Comment (pi0) [inline src/presets/firebase/preset.ts:37]: I believe it is more safe that we throw an error here as it is a wrong build with changing.

File: src/presets/firebase/preset.ts [hunk 2]
```diff
@@ -33,6 +33,12 @@ const firebase = defineNitroPreset(
         nitro.options.appConfig.nitro = nitro.options.appConfig.nitro || {};
         nitro.options.appConfig.nitro.firebase = nitro.options.firebase;
 
+        if(nitro.options.firebase?.serverFunctionName.includes('-')){
```
Comment (pi0) [inline src/presets/firebase/preset.ts:36]: You can use [this util](https://github.com/unjs/knitwork#gensafevariablenamename) to check if name is not equal after normalizing, it is an issue.

Comment (jpsc) [reply src/presets/firebase/preset.ts]: So you are suggesting no normalising just reject the build if `serverFunctionName` is not safe ?

Comment (pi0) [reply src/presets/firebase/preset.ts]: yes.

**Comment by jpsc** [comment]
Not sure how to properly validade this fix.

**Review by pi0 — APPROVED** [review]
Thank you!

**Pull request merged**